statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1312 @@
+
+        Returns a bunch with attributes:
+
+        statistic : float
+            The chi^2 test statistic
+        pvalue : float
+            The p-value of the test statistic
+        df : integer
+            The degrees of freedom of the reference distribution
+
+        Notes
+        -----
+        For a 2x2 table this is equivalent to McNemar's test.  More
+        generally the procedure tests the null hypothesis that the
+        marginal distribution of the row factor is equal to the
+        marginal distribution of the column factor.  For this to be
+        meaningful, the two factors must have the same sample space
+        (i.e. the same categories).
+        """
+
+        if self.table.shape[0] < 1:
+            raise ValueError('table is empty')
+        elif self.table.shape[0] == 1:
+            b = _Bunch()
+            b.statistic = 0
+            b.pvalue = 1
+            b.df = 0
+            return b
+
+        method = method.lower()
+        if method not in ["bhapkar", "stuart_maxwell"]:
+            raise ValueError("method '%s' for homogeneity not known" % method)
+
+        n_obs = self.table.sum()
+        pr = self.table.astype(np.float64) / n_obs
+
+        # Compute margins, eliminate last row/column so there is no
+        # degeneracy
+        row = pr.sum(1)[0:-1]
+        col = pr.sum(0)[0:-1]
+        pr = pr[0:-1, 0:-1]
+
+        # The estimated difference between row and column margins.
+        d = col - row
+
+        # The degrees of freedom of the chi^2 reference distribution.
+        df = pr.shape[0]
+
+        if method == "bhapkar":
+            vmat = -(pr + pr.T) - np.outer(d, d)
+            dv = col + row - 2*np.diag(pr) - d**2
+            np.fill_diagonal(vmat, dv)
+        elif method == "stuart_maxwell":
+            vmat = -(pr + pr.T)
+            dv = row + col - 2*np.diag(pr)
+            np.fill_diagonal(vmat, dv)
+
+        try:
+            statistic = n_obs * np.dot(d, np.linalg.solve(vmat, d))
+        except np.linalg.LinAlgError:
+            warnings.warn("Unable to invert covariance matrix")
+            b = _Bunch()
+            b.statistic = np.nan
+            b.pvalue = np.nan
+            b.df = df
+            return b
+
+        pvalue = 1 - stats.chi2.cdf(statistic, df)
+
+        b = _Bunch()
+        b.statistic = statistic
+        b.pvalue = pvalue
+        b.df = df
+
+        return b
+
+
+    def summary(self, alpha=0.05, float_format="%.3f"):
+
+        fmt = float_format
+
+        headers = ["Statistic", "P-value", "DF"]
+        stubs = ["Symmetry", "Homogeneity"]
+        sy = self.symmetry()
+        hm = self.homogeneity()
+        data = [[fmt % sy.statistic, fmt % sy.pvalue, '%d' % sy.df],
+                [fmt % hm.statistic, fmt % hm.pvalue, '%d' % hm.df]]
+        tab = iolib.SimpleTable(data, headers, stubs, data_aligns="r",
+                                 table_dec_above='')
+
+        return tab
+
+
+
+class Table2x2(SquareTable):
```

Comment:
a `transpose` method would be useful
given the asymmetry in the interpretation (as in the example in the introduction

should it be in `Table` itself and inherited?